data/exfalso.1
.\" Man page generated from reStructuredText.
.
.TH EXFALSO 1 "" "" ""
.SH NAME
exfalso \- audio tag editor
.
.nr rst2man-indent-level 0
.
.de1 rstReportMargin
\\$1 \\n[an-margin]
level \\n[rst2man-indent-level]
level margin: \\n[rst2man-indent\\n[rst2man-indent-level]]
-
\\n[rst2man-indent0]
\\n[rst2man-indent1]
\\n[rst2man-indent2]
..
.de1 INDENT
.\" .rstReportMargin pre:
. RS \\$1
. nr rst2man-indent\\n[rst2man-indent-level] \\n[an-margin]
. nr rst2man-indent-level +1
.\" .rstReportMargin post:
..
.de UNINDENT
. RE
.\" indent \\n[an-margin]
.\" old: \\n[rst2man-indent\\n[rst2man-indent-level]]
.nr rst2man-indent-level -1
.\" new: \\n[rst2man-indent\\n[rst2man-indent-level]]
.in \\n[rst2man-indent\\n[rst2man-indent-level]]u
..
.SH SYNOPSIS
.sp
\fBexfalso\fP [ \fIdirectory\fP ]
.SH DESCRIPTION
.sp
Ex Falso displays and edits audio metadata tags. Supported formats include
MP3, Ogg Vorbis, FLAC, Musepack (MPC), WavPack, and MOD/XM/IT.
.sp
This manual page is only a short reference for Ex Falso. Complete
documentation is available at
\fI\%https://quodlibet.readthedocs.io/en/latest/guide/\fP\&.
.SH OPTIONS
.sp
Ex Falso may be given a directory to open on the command line.
.SH TIED TAGS
.sp
Many places in Ex Falso allow you to use "tied tags". Tied tags are two tag
names joined together with a "~" like "title~version" or "album~part". Tied
tags result in "nice" displays even when one of the tags is missing; for
example, "title~version" will result in Title \- Version when a version tag
is present, but only Title when one isn\(aqt. You can tie any number of tags
together.
.SH RENAMING FILES
.sp
Ex Falso allows you to rename files based on their tags. In some cases you
may wish to alter the filename depending on whether some tags are present
or missing, in addition to their values. A common pattern might be
.sp
\fB<tracknumber>. <title~version>\fP
.sp
You can use a \(aq|\(aq to only text when a tag is present:
.sp
\fB<tracknumber|<tracknumber>. ><title~version>\fP
.sp
You can also specify literal text to use if the tag is missing by adding
another \(aq|\(aq:
.sp
\fB<album|<album>|No Album> \- <title>\fP
.sp
See \fI\%https://quodlibet.readthedocs.io/en/latest/guide/renaming_files.html\fP\&.
.SH BUGS
.sp
See \fI\%https://github.com/quodlibet/quodlibet/issues\fP for a list of all currently
open bugs and feature requests.
.SH AUTHORS
.sp
Joe Wreschnig and Michael Urman are the primary authors of Ex Falso.
.SH SEE ALSO
.nf
\fI\%https://quodlibet.readthedocs.io/en/latest/guide/\fP,
\fI\%https://quodlibet.readthedocs.io/en/latest/guide/faq.html\fP
.fi
.sp
.\" Generated by docutils manpage writer.
.


NEWS.rst
.. _release-4.3.0:

4.3.0 (2020-02-22) - Smaug-like figures, lurking on our planet filled with hoarded data
---------------------------------------------------------------------------------------

Packaging Changes:
  * Various installed files have been renamed
    ("exfalso" -> "io.github.quodlibet.ExFalso") to work better with Flatpak
  * zsh completion file installation location changed to site-functions :pr:`3300`
  * Installs a new bash completion file :pr:`3126` :pr:`3128`
  * Windows: Moved everything from 32 bit to 64 bit. This means QL will no longer work with 32 bit Windows.

Translations:
  * Update brazilian portuguese translations :pr:`3123` (:user:`Hugo Gualandi <hugomg>`)
  * Bulgarian translation fixes :pr`3147` (:user:`cybercop-montana`)
  * Update Hebrew translation :pr:`3164` :pr:`3274` (:user:`Avi Markovitz <avma>`)
  * French translation update :pr:`3183` (:user:`Bundy01`)
  * Update German translation (:user:`Till Berger <Mellthas>`)
  * Various translation related spelling/formatting/text improvements :pr:`3208` :pr:`3207` :pr:`3206` :pr:`3214` :pr:`3215` :pr:`3218` :pr:`3219` (:user:`Till Berger <Mellthas>`)
  * New Serbian translation :pr:`3245` (:user:`leipero`)
  * Update Finnish translation :pr:`3199` (:user:`Kristian Laakkonen <krisse7>`)

General:
  * Simplify launcher for macOS :pr:`3101` (:user:`a-vrma`)
  * Add original date sort option to album browser :pr:`3103` (:user:`Ruud van Asseldonk <ruuda>`)
  * Option for multiple queries in Search Browser :pr:`3082` (:user:`blimmo`)
  * Improved `VGM` Metadata Tag Parsing (GD3 Support) :pr:`3100` (:user:`Eoin O'Neill <Eoin-ONeill-Yokai>`)
  * cover: Always scale down to parent window size :pr:`3114` (:user:`Fredrik Strupe <frestr>`)
  * tags: Support loading lyrics from 'unsyncedlyrics' tag :bug:`3115` (:user:`Fredrik Strupe <frestr>`)
  * browsers: Focus album list on album filtering :bug:`3122` (:user:`Fredrik Strupe <frestr>`)
  * Add bash completion :pr:`3126` :pr:`3128` (:user:`Arnaud Rebillout <elboulangero>`)
  * Docs fixes :pr:`3133` :pr:`3192` (:user:`Petko Ditchev <petko10>`, :user:`CreamyCookie`)
  * player: Make external volume cubic by default :pr:`3151` (:user:`Fredrik Strupe <frestr>`)
  * desktop: Accept all selected files when opened from file browser :bug:`3159` (:user:`Fredrik Strupe <frestr>`)
  * Tracks without track number are now sorted before first track :pr:`3196` (:user:`CreamyCookie`)
  * Add option for ignoring characters in queries :pr:`3221` (:user:`blimmo`)
  * Disable the file trash support under flatpak for now :bug:`3093`
  * zsh completion improvements :pr:`3300` (:user:`Matthew Martin <phy1729>`)
  * Add poetry support :pr:`3306` (:user:`Nick Boultbee <declension>`)

Plugins:
  * Import metadata plugin: Fix file renaming :bug:`3071` (:user:`Fredrik Strupe <frestr>`)
  * Scale player volume properly in mpris2 API :pr:`3098` (:user:`luk1337`)
  * mpris: Drop MPRIS1 support :pr:`3102`
  * Add new Banshee import plugin :pr:`3141` (:user:`Phidica`)
  * Library update plugin: Update on file modifications :bug:`3168` (:user:`Fredrik Strupe <frestr>`)
  * Add "Record Stream" to default custom commands plugin :bug:`1617` (:user:`CreamyCookie`)
  * Custom Commands: Don't modify command when using parameters :bug:`3185` :pr:`3232` (:user:`CreamyCookie`)
  * Import/export plugin: accept full filenames when renaming :pr:`3282` (:user:`Fredrik Strupe <frestr>`)
  * acoustid: fix queries always returning "unknown" :bug:`3269`
  * Fix embed images plugin submenu not showing sometimes :pr`3303` (:user:`Nick Boultbee <declension>`)


.. _release-4.2.1:

4.2.1 (2018-12-26) - DO NOT WEAR THE HAT UNDER ANY CIRCUMSTANCES
----------------------------------------------------------------

Translations:
  * Hebrew translation update :bug:`3027` (:user:`Avi Markovitz <avma>`)
  * German translation update :pr:`3036` (:user:`Till Berger <Mellthas>`)

Fixes:
  * Fix freezes when opening the shortcuts window with i3wm
    :bug:`3055` (:user:`Fredrik Strupe <frestr>`)
  * xinebe: Fix error when pausing a non-local stream :bug:`3057`
  * Fix mmkeys error (preventing QL to start) when mate-settings-daemon is
    running outside of a mate session :bug:`3056`
  * Fix some panels/docks not being able to match the app with the desktop
    file :bug:`3029`
  * Migrate metadata plugin fixes
    :bug:`3070` (:user:`Fredrik Strupe <frestr>`)


.. _release-4.2.0:

4.2.0 (2018-10-31) - staffed by the living
------------------------------------------

Packaging Changes:
  * **gettext 0.19.8** required (was 0.15)
  * **intltool** no longer required

Translations:
  * Finnish translation update
    :pr:`2894` (:user:`Kristian Laakkonen <krisse7>`)
  * Russian translation update :pr:`2965` (:user:`Arkadiy Illarionov <qarkai>`)
  * Danish translation update :pr:`2983` (:user:`scootergrisen`)
  * Polish translation update :pr:`3015` (:user:`Piotr Drąg <piotrdrag>`)

Codebase:
  * Port lots of dbus related code from python-dbus to GDBus
    :pr:`2876` :pr:`2879` :pr:`2885` :pr:`2887` :pr:`2886` :pr:`2901`
    :pr:`2903` :pr:`2904` :pr:`2905` :pr:`2917` :pr:`2918` :pr:`2925`
    (:user:`Arkadiy Illarionov <qarkai>`)
  * Ported from intltool to gettext
  * CI: moved from appveyor to azure-pipelines for Windows testing
  * CI: Add junit test reporting :pr:`2960`
    (:user:`Nick Boultbee <declension>`)
  * Various test suite fixes for NixOS :bug:`2820`
    (:user:`Jan Tojnar <jtojnar>`)
  * Removed most Python 2 compatibility code :pr:`2957`
  * Add a Pipefile for pipenv :pr:`2977`
  * Various minor Python 3.7 compatibility fixes

General:
  * queue: Add option to keep songs after playing them
    :pr:`2865` (:user:`Fredrik Strupe <frestr>`)
  * queue: Add option to not play songs from the queue
    :pr:`2865` (:user:`Fredrik Strupe <frestr>`)
  * Fix non-deterministic ordering of album list and cover grid browsers

quodlibet/exfalso.py
# Copyright 2004-2005 Joe Wreschnig, Niklas Janlert
#           2012 Christoph Reiter
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.

import os

from senf import fsnative, argv as sys_argv

from quodlibet import _
from quodlibet import app
from quodlibet import util
from quodlibet import const
from quodlibet import config


def main(argv=None):
    if argv is None:
        argv = sys_argv

    import quodlibet

    config_file = os.path.join(quodlibet.get_user_dir(), "config")
    quodlibet.init(config_file=config_file)

    from quodlibet.qltk import add_signal_watch
    add_signal_watch(app.quit)

    opts = util.OptionParser(
        "Ex Falso", const.VERSION,
        _("an audio tag editor"), "[%s]" % _("directory"))

    argv.append(os.path.abspath(fsnative(u".")))
    opts, args = opts.parse(argv[1:])
    args[0] = os.path.realpath(args[0])

    app.name = "Ex Falso"
    app.description = _("Audio metadata editor")
    app.id = "io.github.quodlibet.ExFalso"
    app.process_name = "exfalso"
    quodlibet.set_application_info(app)

    import quodlibet.library
    import quodlibet.player
    app.library = quodlibet.library.init()
    app.player = quodlibet.player.init_player("nullbe", app.librarian)
    from quodlibet.qltk.songlist import PlaylistModel
    app.player.setup(PlaylistModel(), None, 0)
    pm = quodlibet.init_plugins()
    pm.rescan()

    from quodlibet.qltk.exfalsowindow import ExFalsoWindow
    dir_ = args[0]
    app.window = ExFalsoWindow(app.library, dir_)
    app.window.init_plugins()

    from quodlibet.util.cover import CoverManager
    app.cover_manager = CoverManager()
    app.cover_manager.init_plugins()

    from quodlibet import session
    session_client = session.init(app)

    quodlibet.enable_periodic_save(save_library=False)
    quodlibet.run(app.window)
    quodlibet.finish_first_session("exfalso")
    config.save()

    session_client.close()

    util.print_d("Finished shutdown.")
